TFNetworkRecLayer.py

```diff
@@ -1435,7 +1435,7 @@ def get_choice_source_batches():
         end_template = self.layer_data_templates["end"]
         needed_outputs.add("end")
         assert tf.as_dtype(end_template.output.dtype) is tf.bool
-        assert end_template.output.batch_shape == (None,)  # (batch*beam,)
+        assert end_template.output.batch_shape == (None,) or end_template.output.batch_shape == (None,1)  # (batch*beam,)
```

Comment:
Why do you need that?
Also, there should be a space after `,` (please use PyCharm, it will tell you PEP8 / Python code conventions).